src/Microsoft.SqlTools.ServiceLayer/EditData/EditDataService.cs

```diff
@@ -0,0 +1,328 @@
+
+                // Setup a callback for when the query failed to be created
+                Func<string, Task> queryCreateFailureCallback = requestContext.SendError;
+
+                // Setup a callback for when the query completes execution successfully
+                Query.QueryAsyncEventHandler queryCompleteSuccessCallback = async query =>
+                {
+                    EditSessionReadyParams readyParams = new EditSessionReadyParams
+                    {
+                        OwnerUri = initParams.OwnerUri
+                    };
+
+                    try
+                    {
+                        // Validate the query for a session
+                        ResultSet resultSet = Session.ValidateQueryForSession(query);
+
+                        // Get a connection we'll use for SMO metadata lookup (and committing, later on)
+                        DbConnection conn = await connectionService.GetOrOpenConnection(initParams.OwnerUri, ConnectionType.Edit);
+                        var metadata = metadataFactory.GetObjectMetadata(conn, resultSet.Columns, 
+                            initParams.ObjectName, initParams.ObjectType);
+
+                        // Create the session and add it to the sessions list
+                        Session session = new Session(resultSet, metadata);
+                        if (!ActiveSessions.TryAdd(initParams.OwnerUri, session))
+                        {
+                            throw new InvalidOperationException("Failed to create edit session, session already exists.");
+                        }
+                        readyParams.Success = true;
+                    }
+                    catch (Exception)
+                    {
+                        // Request that the query be disposed
+                        await queryExecutionService.InterServiceDisposeQuery(initParams.OwnerUri, null, null);
+                        readyParams.Success = false;
+                    }
+
+                    // Send the edit session ready notification
+                    await requestContext.SendEvent(EditSessionReadyEvent.Type, readyParams);
+                };
+
+                // Setup a callback for when the query completes execution with failure
+                Query.QueryAsyncEventHandler queryCompleteFailureCallback = query =>
+                {
+                    EditSessionReadyParams readyParams = new EditSessionReadyParams
+                    {
+                        OwnerUri = initParams.OwnerUri,
+                        Success = false
+                    };
+                    return requestContext.SendEvent(EditSessionReadyEvent.Type, readyParams);
+                };
+
+                // Put together a query for the results and execute it
+                ExecuteStringParams executeParams = new ExecuteStringParams
+                {
+                    Query = $"SELECT * FROM {SqlScriptFormatter.FormatMultipartIdentifier(initParams.ObjectName)}",
```

Comment:
It's a C# 6 feature, string interpolation. Basically a string literal that has a $ in front of the quotation mark will include anything inside `{}` in the string. It's pretty straight forward and I don't think it's unclear what's going on, especially considering this is quite similar to how Ruby, PowerSell and other languages offer string interpolation.